src/components/config/config.ts

```diff
@@ -84,14 +85,35 @@ export function getActiveKubeconfig(): string {
 
 // Functions for working with tool paths
 
-export function getToolPath(host: Host, tool: string): string | undefined {
-    return host.getConfiguration(EXTENSION_CONFIG_KEY)[toolPathKey(tool)];
+export function getToolPath(host: Host, shell: Shell, tool: string): string | undefined {
```

Comment:
instead of passing the shell down to here, would it be acceptable to pass the platform directly? or make it a property available on `host`?

Reply:
I'm torn.  On the one hand, yes, all the function _needs_ is the platform, so why pass in a monstrous `Shell`?  On the other hand, _everywhere_ this is called, it has a `Shell` handy but it usually _hasn't_ pulled out the `platform`, so _all_ the call sites would be polluted with a call to `.platform()`.

Not keen on putting it on `Host` as that is mostly wrapping the VS Code environment.  Though the combination of a `Host` and a `Shell` is so common we could wrap that up into a single interface, something like `ExtensionContext` though that's already taken **grin**.

I am probably overthinking this.  _sigh_